### PR TITLE
fix: Assertion methods dont cleanup when failing

### LINKED ISSUE #38: Sinco Doesnt Clean Up After Failed Assertion Method
## Summary

A clear and concise summary of the bug.

## Steps To Reproduce The Bug

1. Create a test using this code: https://staging.drash.io/sinco/v1.x/#/tutorials/custom-assertions
2. Replace whatever is in `assertSee` with anything else
3. Run test, expect to fail
4. Run test again, it will now hang

## Expected Behavior

It should not hang after running after a failed test

## Suggested Solution(s)

Assertion methods should cleanup on the event that an assertion will fail

### PR DESCRIPTION
**Pull request by ebebbington**
Fixes #38 

## Summary

  * Assertion method now cleaup if they. fail
  * Added tests for the above
  * Added test that replicates thee test on the custom_assertions page

### CODE REVIEW

File: src/headless_browser.ts [hunk 1]
```diff
@@ -208,6 +213,9 @@ export class HeadlessBrowser {
       };
     };
     const actualUrl = res.root.documentURL;
+    if (actualUrl !== expectedUrl) { // Before we know the test will fail, close everything
+      await this.done();
```
Comment (saragee3) [inline src/headless_browser.ts:217]: should we throw an error at all?

File: src/headless_browser.ts [hunk 2]
```diff
@@ -223,7 +231,11 @@ export class HeadlessBrowser {
     });
     this.checkForErrorResult((res as DOMOutput), command);
     // Tried and tested, and `result` is `{result: { type: "boolean", value: false } }`
-    const exists = ((res as DOMOutput).result as SuccessResult).value;
+    const exists = ((res as DOMOutput).result as SuccessResult)
+      .value as boolean;
+    if (exists !== true) {
+      await this.done();
```
Comment (saragee3) [inline src/headless_browser.ts:237]: same question here about throwing an error. does this just silently close then?

File: src/headless_browser.ts (hunk 1 above)
Comment (ebebbington) [reply src/headless_browser.ts:217]: How ddo you mean throw an error? Oh are you saying we should esentially turn this method into just `getUrl()`?, then the user would do:

```
const url = await Sinco.getUrl()
assertEquals(url, ...) // <-- The USER does the assertions
```

But atm, all that method does is just makes an assertions really

File: src/headless_browser.ts (hunk 2 above)
Comment (ebebbington) [reply src/headless_browser.ts:237]: Yeah, the `done()` method just closes the ws conn and stops the subprocess - it's essentially saying "If we can see that the assertion is going to fail, cleanup Sinco"

Comment (saragee3) [reply src/headless_browser.ts:237]: ohh i see `assertSee` and `assertUrlIs` are just for testing... should they even be in this class? 🤔

**Review by saragee3 — APPROVED** [review]
lgtm! the testing methods still feel weird to me (in the same class as the main HeadlessBrowser) but we can address that some other time. we're so close!

File: src/headless_browser.ts (hunk 2 above)
Comment (ebebbington) [reply src/headless_browser.ts:237]: Yeah that would make sense tbh, not entirely sure how to architect this but we can make an issue for it and discuss it?

Comment (ebebbington) [reply src/headless_browser.ts:237]: Ah just seen:

> lgtm! the testing methods still feel weird to me (in the same class as the main HeadlessBrowser) but we can address that some other time. we're so close!

Ill make an issue for it then

**Pull request merged**